base/iterators.jl

```diff
@@ -1171,10 +1171,10 @@ function length(itr::PartitionIterator)
     return cld(l, itr.n)
 end
 
-function iterate(itr::PartitionIterator{<:AbstractRange}, state=1)
-    state > length(itr.c) && return nothing
+function iterate(itr::PartitionIterator{<:AbstractVector}, state = firstindex(itr.c))
+    state > lastindex(itr.c) && return nothing
     r = min(state + itr.n - 1, length(itr.c))
-    return @inbounds itr.c[state:r], r + 1
+    return @inbounds view(itr.c, state:r), r + 1
 end
 
 function iterate(itr::PartitionIterator{<:AbstractArray}, state=1)
```

Comment:
Looks like the `AbstractArray` method can be improved, too?

Reply:
Hmmm, actually, no need. Because here it's using LinearIndexing rules, which always assume 1-based.